Fails for when OPAMROOT is set to something other than the default directory

When `OPAMROOT` is set to something other than the default directory, `opam-monorepo` fails:

```
[ERROR] Opam has not been initialised, please run `opam init'
opam-monorepo: internal error, uncaught exception:
               OpamStd.OpamSys.Exit(50)
               Raised at file "src/core/opamStd.ml", line 1042, characters 15-29
               Called from file "src/state/opamGlobalState.ml", line 60, characters 4-113
               Called from file "src/state/opamGlobalState.ml", line 185, characters 11-20
               Called from file "cmdliner_term.ml", line 25, characters 19-24
               Called from file "cmdliner.ml", line 33, characters 27-34
               Called from file "cmdliner.ml", line 117, characters 32-39
make: *** [Makefile:61: deps-pull] Error 125
```

It doesn't seem like monorepo is initialising the opam state in the same way opam does, so the `OPAMROOT` variable is never read.

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,3 +1,17 @@
+## unreleased
+
+### Added
+
+### Changed
+
+### Deprecated
+
+### Fixed
+
+### Removed
+
+### Security
+
 ## 0.2.4
 
 ### Added

--- a/CHANGES.md
+++ b/CHANGES.md
@@ -8,6 +8,9 @@
 
 ### Fixed
 
+- Fix setting `OPAMROOT` to accept non-default paths (#197, #198,
+  @Leonidas-from-XIV)
+
 ### Removed
 
 ### Security

--- a/bin/opam_monorepo.ml
+++ b/bin/opam_monorepo.ml
@@ -25,7 +25,7 @@ let init_opam () =
   OpamFormatConfig.init ();
   OpamCoreConfig.init ~safe_mode:true ();
   OpamRepositoryConfig.init ();
-  OpamStateConfig.init ()
+  OpamStateConfig.init ~root_dir:root ()
 
 let () =
   init_opam ();

--- a/cli/dune
+++ b/cli/dune
@@ -1,4 +1,12 @@
 (library
  (name duniverse_cli)
- (libraries duniverse_lib fmt.cli fmt.tty logs.cli logs.fmt cmdliner lwt.unix
-   dune-build-info ocaml-version))
+ (libraries
+  duniverse_lib
+  fmt.cli
+  fmt.tty
+  logs.cli
+  logs.fmt
+  cmdliner
+  lwt.unix
+  dune-build-info
+  ocaml-version))

--- a/lib/dune
+++ b/lib/dune
@@ -1,4 +1,16 @@
 (library
  (name duniverse_lib)
- (libraries base bos fmt logs lwt.unix ocaml-version opam-0install
-   opam-file-format opam-format sexplib stdext threads uri))
+ (libraries
+  base
+  bos
+  fmt
+  logs
+  lwt.unix
+  ocaml-version
+  opam-0install
+  opam-file-format
+  opam-format
+  sexplib
+  stdext
+  threads
+  uri))

